modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -376,15 +376,35 @@ const getRootBlockName = (settings: DomParserSettings, args: ParserArgs) => {
   }
 };
 
-const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser => {
+const DomParser = (settings: DomParserSettings = {}, schema = Schema()): DomParser => {
   const nodeFilters: Record<string, ParserFilterCallback[]> = {};
   const attributeFilters: ParserFilter[] = [];
 
-  settings = settings || {};
-  settings.validate = 'validate' in settings ? settings.validate : true;
-  settings.root_name = settings.root_name || 'body';
+  // Apply setting defaults
+  const defaultedSettings = {
+    validate: true,
+    root_name: 'body',
+    ...settings
+  };
+
+  const parser = new DOMParser();
+  const purify = setupPurify(defaultedSettings, schema);
+
+  const parseAndSanitizeWithContext = (html: string, rootName: string, format: string = 'html') => {
+    const mimeType = format === 'xhtml' ? 'application/xhtml+xml' : 'text/html';
```

Comment:
Will we ever use this xhtml parser part or should we just hard code it to be `text/html`?

Reply:
We currently support both XHTML and HTML in TinyMCE 5 so we need to support both unless someone higher up says we can remove both. Given our deadlines we likely aren't going to be able to attempt that for 6.x.

Reply:
Yes, but a html5 parser can parse xhtml will this not force the document to be required to be a XHTML valid document don't think we ever supported that strict ruleset.

The older parser would just parse things like CDATA and PI elements however it would still parse non XML as `<ul><li>foo</ul>` a strict XHTML parser will not and produces a document with a parsererror element.

So it's more about do we need to retain CDATA and PI elements in the parsing step. I don't think we need to and if we do we could preprocess them so they get parsed properly as comments or something that we later can convert back into CDATA and PI elements int he ast tree.

But this is only for people using the parser directly this will not be used if they configure the output format to be xhtml right. So the risk of this new strict parsing behavior causing issues is low?

Reply:
> But this is only for people using the parser directly this will not be used if they configure the output format to be xhtml right. So the risk of this new strict parsing behavior causing issues is low?

Yes, this would only be used when using the API directly... tbh I believe this was only ever used in the `fullpage` plugin so it's something we should look at removing in 7.0.